ibmcloud-janitor/resources/serviceid.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	identityv1 "github.com/IBM/platform-services-go-sdk/iamidentityv1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/boskos/common/ibmcloud"
+	"sigs.k8s.io/boskos/ibmcloud-janitor/account"
+	"sigs.k8s.io/boskos/ibmcloud-janitor/utils"
+)
+
+type APIKey struct {
+	serviceIDName string
+	value         *string
+}
+
+func getAPIkeyName() (string, error) {
+	n, err := utils.RandomNumber()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("test-key-%d", n), nil
+}
```

Comment:
wondering if we can use some k8s utilities to generate the random string here.

Reply:
Changed to use k8s utility.